### PR TITLE
MINOR: Improve javadoc of user-customizable metrics API

### PR DESCRIPTION
**Pull request by cadonna**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java [hunk 1]
```diff
@@ -170,29 +181,35 @@ void recordThroughput(final Sensor sensor,
...
      *
      * @param name           name of the sensor.
      * @param recordingLevel the recording level (e.g., INFO or DEBUG) for this sensor
      * @return The added sensor.
+     * @see #addRateTotalSensor(String, String, String, RecordingLevel, String...)
+     * @see #addLatencyRateTotalSensor(String, String, String, RecordingLevel, String...)
+     * @see #removeSensor(Sensor)
      */
     Sensor addSensor(final String name,
                      final Sensor.RecordingLevel recordingLevel);
```
Comment (cadonna) [inline streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java:197]: Isn't this method subsumed by the method on line 214? Shall we deprecate this method?

**Comment by cadonna** [comment]
Call for review: @guozhangwang @bbejeck

File: streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java (hunk 1 above)
Comment (bbejeck) [reply streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java:197]: Looks like we could deprecate it, but it's not listed for deprecation in the KIP.  I'll defer to @guozhangwang on this one.

**Review by bbejeck — APPROVED** [review]
Thanks for the PR @cadonna.  LGTM

File: streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java:197]: I think these are supposed to be two overloaded functions but yeah I agree that the overloading here seems not worth much :P I think it's okay just to leave as is.

**Review by guozhangwang — APPROVED** [review]
LGTM. Feel free to merge @bbejeck

**Comment by bbejeck** [comment]
Merged #7810 into trunk.

**Comment by bbejeck** [comment]
Thanks for the javadoc improvement @cadonna!

**Pull request merged**